src/parser/ultimate_guitar_parser.js

```diff
@@ -0,0 +1,105 @@
+import ChordSheetParser from './chord_sheet_parser';
+import { CHORUS, NONE, VERSE } from '../constants';
+import {
+  END_OF_CHORUS,
+  END_OF_VERSE,
+  START_OF_CHORUS,
+  START_OF_VERSE,
+} from '../chord_sheet/tag';
+
+const noop = () => {};
+const logger =
+  process && process.env === 'development'
+    ? console
+    : {
+      debug: noop,
+      error: noop,
+      info: noop,
+    };
+
+const VERSE_LINE_REGEX = /^\[Verse.*\]/;
+const CHORUS_LINE_REGEX = /^\[Chorus\]/;
+const OTHER_METADATA_LINE_REGEX = /^\[([^\]]+)\]/;
+
+const sectionTypeToStartTag = {
+  [CHORUS]: START_OF_CHORUS,
+  [VERSE]: START_OF_VERSE,
+};
+
+const sectionTypeToEndTag = {
+  [CHORUS]: END_OF_CHORUS,
+  [VERSE]: END_OF_VERSE,
+};
+
+/**
+ * Parses an Ultimate Guitar chord sheet with metadata
+ */
+export default class UltimateGuitarParser extends ChordSheetParser {
+  parse(ultimateGuitarChordSheet) {
+    this.currentSectionType = NONE;
+    const song = super.parse(ultimateGuitarChordSheet);
+    this.endCurrentSection();
+    return song;
+  }
+
+  startSection(type) {
+    this.endCurrentSection();
+
+    logger.debug(`Starting ${type} section`);
+
+    if (type in sectionTypeToStartTag) {
+      const startTag = sectionTypeToStartTag[type];
+      logger.debug(`Adding section start tag ${startTag}`);
+      this.song.addTag(startTag);
```

Comment:
@bkeepers Those tags are used to attach css classes for verse and chorus to html tags. They are not displayed currently, you could subclass the formatter. Out of curiousity: do you expect it to be rendered because another app does that?

Reply:
> Out of curiousity: do you expect it to be rendered because another app does that?

No, not necessarily. My point was just that the parser is lossy. I didn't realize it was adding a css class, so that helps, but you still lose verse numbers.

Reply:
Ok clear. Yeah, it might make sense to store indices per section and expose them for use by the formatters.